roles/deploy/tasks/share.yml

```diff
@@ -2,25 +2,41 @@
 - include: "{{ deploy_share_before | default('../hooks/example.yml') }}"
   tags: deploy-share-before
 
-- name: Ensure shared sources are present
+- name: Ensure shared sources are present -- directories
+  file:
+    path: "{{ deploy_helper.shared_path }}/{{ project_shared_file | ternary(item.src | dirname, item.src) }}"
+    state: "{{ project_shared_file | ternary('directory', item.type | default('directory')) | lower }}"
+    mode: "{{ project_shared_file | ternary('0755', item.mode | default('0755')) }}"
+  with_items: "{{ project_shared_children }}"
+
+- name: Ensure shared sources are present -- files
   file:
     path: "{{ deploy_helper.shared_path }}/{{ item.src }}"
-    state: "{{ item.type | default('directory') }}"
-    mode: "{{ item.mode | default('0755') }}"
+    state: touch
+    mode: "{{ item.mode | default('0644') }}"
+  with_items: "{{ project_shared_children }}"
+  when: project_shared_file
+
+- name: Ensure parent directories for shared paths are present
+  file:
+    path: "{{ deploy_helper.new_release_path }}/{{ item.path | default('') | dirname }}"
+    state: directory
   with_items: "{{ project_shared_children }}"
 
 - name: Ensure shared paths are absent
   file:
-    path: "{{ deploy_helper.new_release_path }}/{{ item.path }}"
+    path: "{{ deploy_helper.new_release_path }}/{{ item.path  | default('') }}"
     state: absent
   with_items: "{{ project_shared_children }}"
+  when: item.type | default('directory') | lower != 'absent'
 
 - name: Create shared symlinks
   file:
-    path: "{{ deploy_helper.new_release_path }}/{{ item.path }}"
+    path: "{{ deploy_helper.new_release_path }}/{{ item.path | default('') }}"
```

Comment:
Shouldn't we enforce the `path` here? Items set to `absent` aren't included here anyway.